app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageViewModel.kt

```diff
@@ -22,4 +37,45 @@ class ProfilePageViewModel @Inject constructor(
         }
         _onLogOutButtonClicked.call()
     }
+
+    val userData = MutableLiveData<MainUserInfo>()
+
+    val loading = mutableStateOf(false)
+
+    fun getUserInfo() {
+        loading.value = true
+        getUserData
+            .execute()
+            .doOnSubscribe {
+                isLoadingLiveData.postValue(true)
+            }
+            .doFinally {
+                isLoadingLiveData.postValue(false)
+            }
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                userData.value = it
+                loading.value = false
+            }, {
+                handleError(it)
+            })
+            .run(compositeDisposable::add)
+    }
+
+    private fun writeUserImageUri(uri: Uri) {
+        writeUserImageUri
+            .execute(uri)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({}, {})
```

Comment:
why success and error methods are empty?

Reply:
done